File: OpenDentBusiness/Data Interface/EmailAttaches.cs
using System;
using System.Collections;
using System.Data;
using System.Reflection;

namespace OpenDentBusiness{
	///<summary></summary>
	public class EmailAttaches{

		public static long Insert(EmailAttach attach) {
			if(RemotingClient.RemotingRole==RemotingRole.ClientWeb) {
				attach.EmailAttachNum=Meth.GetInt(MethodBase.GetCurrentMethod(),attach);
				return attach.EmailAttachNum;
			}
			if(PrefC.RandomKeys) {
				attach.EmailAttachNum=ReplicationServers.GetKey("emailattach","EmailAttachNum");
			}
			string command= "INSERT INTO emailattach (";
			if(PrefC.RandomKeys) {
				command+="EmailAttachNum,";
			}
			command+="EmailMessageNum, DisplayedFileName, ActualFileName) VALUES(";
			if(PrefC.RandomKeys) {
				command+="'"+POut.PLong(attach.EmailAttachNum)+"', ";
			}
			command+=
				 "'"+POut.PLong(attach.EmailMessageNum)+"', "
				+"'"+POut.PString(attach.DisplayedFileName)+"', "
				+"'"+POut.PString(attach.ActualFileName)+"')";
			if(PrefC.RandomKeys) {
				Db.NonQ(command);
			}
			else {
				attach.EmailAttachNum=Db.NonQ(command,true);
			}
			return attach.EmailAttachNum;
		}


	}

	


}

File: java/OpenDentBusiness/EmailAttaches.java
//
// Translated by CS2J (http://www.cs2j.com): 2/15/2016 8:00:43 PM
//

package OpenDentBusiness;

import OpenDentBusiness.EmailAttach;
import OpenDentBusiness.Meth;
import OpenDentBusiness.POut;
import OpenDentBusiness.RemotingClient;
import OpenDentBusiness.RemotingRole;

/**
* 
*/
public class EmailAttaches   
{
    public static long insert(EmailAttach attach) throws Exception {
        if (RemotingClient.RemotingRole == RemotingRole.ClientWeb)
        {
            attach.EmailAttachNum = Meth.GetLong(MethodBase.GetCurrentMethod(), attach);
            return attach.EmailAttachNum;
        }
         
        return Crud.EmailAttachCrud.Insert(attach);
    }

    public static List<EmailAttach> getForEmail(long emailMessageNum) throws Exception {
        if (RemotingClient.RemotingRole == RemotingRole.ClientWeb)
        {
            return Meth.<List<EmailAttach>>GetObject(MethodBase.GetCurrentMethod(), emailMessageNum);
        }
         
        String command = "SELECT * FROM emailattach WHERE EmailMessageNum=" + POut.long(emailMessageNum);
        return Crud.EmailAttachCrud.SelectMany(command);
    }

}


